### PR TITLE
Fix/82-83 zero testdata update bug and parameter update bug in testdata

### PR DESCRIPTION
**Pull request by poojan-2601**

### CODE REVIEW

File: app/routes/payloads.py [hunk 1]
```diff
@@ -73,8 +73,8 @@ def create_payloads():
         try:
             for exp_outcome in expected_outcome:
                 if not (exp_outcome['expected_outcome'] is not None and is_expected_outcome_valid(exp_outcome['expected_outcome'])):
-                    raise Exception(
-                        'You cannot pass empty expected outcome in the payload')
+                    logging.info(f"POST request to create payload failed as empty expected outcome was provided in the payload")
+                    return jsonify({"error": "You cannot pass empty expected outcome in the payload"})
```
Comment (tanmaythole) [inline app/routes/payloads.py]: status code?

File: app/routes/testcases.py [hunk 2]
```diff
@@ -146,6 +146,9 @@ def update_testcase():
                 testcase.teststeps.clear()
                 testcase.testdatas.clear()
                 for test_step in teststeps:
+                    if not ((type(test_step.get('teststep')) is int) and (type(test_step.get('testdata') is list) and len(test_step.get('testdata')) > 0)):
+                        logging.info(f"PUT request to update testcases failed due as faulty data provided in teststep or zero testdata provided for testcase")
+                        return jsonify({"error" : "faulty data provided in teststep or zero testdata provided for testcase"}), 400
```
Comment (tanmaythole) [inline app/routes/testcases.py]: Can we change the error message so that the user will understand?

**Review by tanmaythole — CHANGES_REQUESTED** [review]

File: app/routes/payloads.py (hunk 1 above)
Comment (poojan-2601) [reply app/routes/payloads.py]: done

File: app/routes/testcases.py (hunk 2 above)
Comment (poojan-2601) [reply app/routes/testcases.py]: done

**Review by tanmaythole — APPROVED** [review]

**Pull request merged**